# üß¨ Needleman-Wunsch Global Alignment Matrix

This repository contains an implementation of the **Needleman-Wunsch algorithm**, a classic **Dynamic Programming** approach used for **Global Sequence Alignment** in bioinformatics.  

Global alignment ensures that **two sequences are compared from start to end**, considering **matches, mismatches, and gaps** to find the optimal alignment.

---

## üìå Features
- Implements the **Needleman-Wunsch Algorithm** in Python  
- Supports customizable **scoring scheme**:
  - Match reward  
  - Mismatch penalty  
  - Gap penalty  
- Builds the **alignment matrix** step by step  
- Can be extended to perform **traceback** and output aligned sequences  

<img width="663" height="528" alt="confusion-metrix" src="https://github.com/user-attachments/assets/aa9e6716-d487-4e46-8ebe-db5f52a2b595" />


<img width="663" height="528" alt="arrow-confusion-metrix" src="https://github.com/user-attachments/assets/bf696f1f-660f-4f4f-86c6-94ef9112bf08" />



---

## Mubashir Ali - Founder @ Code with Bismillah | Aspiring Bioinformatics & Data Science Professional | Bridging Biology & Data | Researcher | Genomics, Machine Learning, AI | Python, R, Bioinformatics Tools | QAU

## ‚öôÔ∏è Installation
Clone the repository:
```bash
git clone https://github.com/mubashir1837/Needleman-Wunsch-Global-Alignment-Matrix.git
cd Needleman-Wunsch-Global-Alignment-Matrix


